src/runtime/tests/manifest-test.ts

```diff
@@ -3659,6 +3659,130 @@ resource SomeName
       `), `Type variable ~a does not contain field 'foo'`);
     });
 
+    it('fails to parse concrete types with inline schema field of `*`', async () => {
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {*}
+      `), `\
+Post-parse processing error caused by 'undefined' line 3.
+Only type variables may have '*' fields.
+              data: reads {*}
+                          ^^^`);
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {name: Text, *}
+      `), `\
+Post-parse processing error caused by 'undefined' line 3.
```

Comment:
I don't think this is a good error message to allow into production, it doesn't give the user anything to fix and also doesn't tell the user that there was an internal problem. I think it means that your parser code is constructing data that doesn't pass the `checkNormal` function's requirements.

Please fix that up before landing this.

Reply:
The error message is below, it is "Only type variables may have '*' fields."

"Post-parse processing error caused by 'undefined' line 3." is standard error , `undefined` would be replaced with a file name FWIW.

Reply:
Ah, thanks, misunderstood the error.
Resolved!